owncloud/phoenix
# close #516 search for files

## Description
search for files in files app via ownCloud search.
It uses the search-bar component, which get's enriched with auto-trim functionality in this PR....
Because most search feature's don't accept empty strings as search pattern.

I had to hide LastEdit date & FileSize, because ownCloud does not return them in search.
Is this intended behaviour ?

I've implemented some regex workaround to generate the file extension from name, but i'm not able to differentiate between folder or file; So i use the File icon as fallback, even for folders.
My workaround enables the use of the custom file type icons.

## Related Issue




- Fixes #516

## Motivation and Context
global search is a nice thing


## How Has This Been Tested?



- open Files app
- type 'md'
- see markdown files

## Screenshots
![filesearch](https://user-images.githubusercontent.com/7268689/52169823-a4ed9500-273e-11e9-9ae2-da574076c396.png)

## Types of changes


## Checklist



## Open tasks

Remove leading slash in file path ?

The current UX is miserable, the global search does not change the breadcrumb / path, the local folder search is hidden in filters, etc. We should discuss this further!

@DeepDiver1975 the init bug is still available, as discussed in talk #phoenix room.
Search bar has autofocus, so if you type fast enough, your search will crash.
Do you prefer to use your double-init-workaround or should I open an issue in  [js-owncloud-client](https://github.com/owncloud/js-owncloud-client) ? I prefer the second option.
